pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -429,6 +437,41 @@ func (r *ReconcilePerconaServerMongoDB) Reconcile(request reconcile.Request) (re
 	return rr, nil
 }
 
+func (r *ReconcilePerconaServerMongoDB) compareWithCurrentReplSets(cr *api.PerconaServerMongoDB) error {
+	sfsList := appsv1.StatefulSetList{}
+	if err := r.client.List(context.TODO(), &sfsList,
+		&client.ListOptions{
+			Namespace: cr.Namespace,
+			LabelSelector: labels.SelectorFromSet(map[string]string{
+				"app.kubernetes.io/instance": cr.Name,
+			}),
+		},
+	); err != nil {
+		return errors.Wrap(err, "failed to get statefulset list")
+	}
+
+	appliedNames := make(map[string]struct{}, len(cr.Spec.Replsets))
+	for _, v := range cr.Spec.Replsets {
+		appliedNames[cr.Name+"-"+v.Name] = struct{}{}
+	}
+
+	for _, v := range sfsList.Items {
```

Comment:
Shouldn't we do inverse logic - compare values from the applied set with the incoming changes?
Now it protects against 
```
Applied    New
    a       b
```
But not
```
Applied    New
    a       b
    b
```
Because "b" would be in Applied. But "a" not. And it hasn't checked

Reply:
Oh, sorry, my bad. Wrongly read the code

Reply:
appliedNames is a bit confusing name, it sounds like it a set of already applied on cluster stateful sets